src/sickle.ts

```diff
@@ -399,6 +406,75 @@ class Annotator {
     }
   }
 
+  private visitExterns(node: ts.Node, namespace: string[] = []) {
+    this.currentOutput = this.externsOutput;
+    switch (node.kind) {
+      case ts.SyntaxKind.ModuleDeclaration:
+        let decl = <ts.ModuleDeclaration>node;
+        namespace = namespace.concat([decl.name.text]);
+        this.visitExterns(decl.body, namespace);
+        break;
+      case ts.SyntaxKind.ModuleBlock:
+        let block = <ts.ModuleBlock>node;
+        for (let stmt of block.statements) {
+          this.visitExterns(stmt, namespace);
+        }
+        break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeExternsInterface(<ts.InterfaceDeclaration>node, namespace);
+        break;
+      case ts.SyntaxKind.VariableStatement:
+        for (let decl of(<ts.VariableStatement>node).declarationList.declarations) {
+          this.writeExternsVariable(decl, namespace);
+        }
+        break;
+      default:
+        this.errorUnimplementedKind(node, 'externs generation');
+        break;
+    }
+    this.currentOutput = this.tsOutput;
+  }
+
+  private writeExternsInterface(decl: ts.InterfaceDeclaration, namespace: string[]) {
+    this.emit('/** @record @struct */\n');
```

Comment:
They are too ill-defined to truly "mean" much :) The closest thing I have to a specification are the examples here (which don't specify behavior of @struct on interfaces) :/
https://github.com/google/closure-compiler/wiki/@struct-and-@dict-Annotations
https://github.com/google/closure-compiler/wiki/Structural-Interfaces-in-Closure-Compiler

So the only thing left is poking at the implementation:
https://closure-compiler-debugger.appspot.com/home#input0%3D%252F**%2520%2540record%2520%2540struct%2520*%252F%250Afunction%2520Recursive()%257B%257D%253B%250A%252F**%2520%2540type%2520%257B%253FRecursive%257D%2520*%252F%250ARecursive.prototype.next%253B%250A%250Avar%2520x%2520%253D%2520%257B%257D%250Ax.next%2520%253D%2520x%253B%250A%252F**%2520%2540type%2520%257BRecursive%257D%2520*%252F%250Avar%2520a%2520%253D%2520x%253B%250A%250Aa%255B'foo'%255D%2520%253D%25200%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

Which indeed shows that @struct makes a difference for allowing/disallowing index setters. So good call to keep both.